master/pkg/searcher/random.go

```diff
@@ -83,15 +83,17 @@ func (s *randomSearch) trialExitedEarly(
 	ctx context, requestID model.RequestID, exitedReason model.ExitedReason,
 ) ([]Operation, error) {
 	s.PendingTrials--
-	if exitedReason == model.InvalidHP || exitedReason == model.InitInvalidHP {
-		var ops []Operation
-		create := NewCreate(ctx.rand, sampleAll(ctx.hparams, ctx.rand), model.TrialWorkloadSequencerType)
-		ops = append(ops, create)
-		ops = append(ops, NewValidateAfter(create.RequestID, s.MaxLength()))
-		ops = append(ops, NewClose(create.RequestID))
-		// We don't increment CreatedTrials here because this trial is replacing the invalid trial.
-		s.PendingTrials++
-		return ops, nil
+	if s.randomSearchState.SearchMethodType == RandomSearch {
```

Comment:
At this point I'm inclined to just pull single out into its own searcher. That way for features like invalid hp the dev would explicitly think through "now what should this do for single search" and bugs like this would be less likely.

But that's secondary to just fixing the bug.

Reply:
```suggestion
	if s.SearchMethodType == RandomSearch {
```